### DISCUSSION TITLE
Converting a complete python script to Tableau/TabPy

### DISCUSSION THREAD

**Discussion by BharSgavi** [tableau/TabPy#540] [General]
Hey everyone! 
I am new to TabPy and am trying to find out if there is a way to import a py file as-is in Tableau.
My python piece is calling multiple codes(in different files) and doing cleansing, some analytics and some visualization. Is there anyway I can use the code and just replace the visualization part with Tableau ?

**Comment by nmannheimer** [comment]
Hi @BharSgavi generally speaking the answer is yes, but it's difficult to answer specifically here without more details. I would look at potentially plugging TabPy into Tableau Prep, using your code to output a CSV you can connect to in Tableau, or trying our new Table Extensions feature in Tableau's 22.3 Beta.
